vaxrank/vaccine_peptide.py

```diff
@@ -54,14 +60,35 @@ def lexicographic_sort_key(self):
         score positive (since we want fewer wildtype epitopes) but the others
         negative (since we want more of them).
         """
-        return (
+        # since we're sorting in decreasing order, numbers which we want
+        # to be larger must have their signs flipped
+        essential_score_tuple = (
+            # Sum of normalized MHC binding affinities of subsequences
             -self.mutant_epitope_score,
-            -self.mutant_protein_fragment.n_alt_reads,
+            # Number of reads supporting the variant
+            -self.mutant_protein_fragment.n_alt_reads
+        )
+        manufacturability_score_tuple = tuple(self.manufacturability_scores)
+        extra_score_tuple = (
+            # Number of reads supporting the particular protein sequence
+            # sequence we're using for this vaccine peptide. Currently
+            # all vaccine peptides are drawn from the same larger sequence
+            # so this score shouldn't change.
             -self.mutant_protein_fragment.n_alt_reads_supporting_protein_sequence,
+            # Minimize the sum of non-mutant MHC binding scores
             self.wildtype_epitope_score,
+            # All else being equal, we prefer to maximize the number of
+            # mutant amino acids
             -self.mutant_protein_fragment.n_mutant_amino_acids,
+            # If nothing else can serve as a tie break then try to center
+            # the mutation in the vaccine peptide.
             -self.mutant_protein_fragment.mutation_distance_from_edge
         )
+        return (
+            essential_score_tuple +
+            manufacturability_score_tuple +
```

Comment:
It seems like there are at least two immunogenic concerns here: (1) minimizing the wildtype epitope score and (2) the unfortunate fact that [TCR contact residue hydrophobicity is a hallmark of immunogenic CD8+ T cell epitopes.](https://www.ncbi.nlm.nih.gov/pubmed/25831525). 

One thing I can imagine doing is splitting the hydrophobicity scores tuple into higher priority binary scores (C-term 7mer GRAVY > 1.5, max kmer GRAVY >1.5) and then a lower priority continuous value that we place below wildtype epitope score. 

What do you think?